### PR TITLE
 feat: adding sso_providers api end-point

### PR DESCRIPTION
**Pull request by akoserwal**
Adding sso provider end-point which will return the urls for the sso used by the data plane clusters

`/api/kafkas_mgmt/v1/sso_providers`

return 
``` {
    "base_url": "https://identity.api.stage.openshift.com",
    "token_url": "https://identity.api.stage.openshift.com/auth/realms/rhoas/protocol/openid-connect/token",
    "jwks": "https://identity.api.stage.openshift.com/auth/realms/rhoas/protocol/openid-connect/certs",
    "valid_issuer": "https://identity.api.stage.openshift.com/auth/realms/rhoas"
}

```

Looking for initial feedback before adding test/improving it

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

**Comment by dhirajsb** [comment]
Couldn't these be part of addon parameters provided to the agent?

**Comment by machi1990** [comment]
> Couldn't these be part of addon parameters provided to the agent?

No, this is the user facing endpoint. The fleetshard will always get the right sso url via the ManagedKafkaCR. With this sso metadata endpoint, we need a way to tell the user where their service accounts are located so that they can correctly authenticate to their Kafka.

File: internal/kafka/internal/handlers/service_accounts.go [hunk 1]
```diff
@@ -171,3 +172,21 @@ func (s serviceAccountsHandler) GetServiceAccountById(w http.ResponseWriter, r *
...
+				TokenUrl: config.TokenEndpointURI,
+				ValidIssuer: config.ValidIssuerURI,
+			}
+
+			return presenters.PresentSsoProvider(&provider), nil
+		},
+	}
+
+	handlers.HandleGet(w, r, cfg)
+}
```
Comment (machi1990) [inline internal/kafka/internal/handlers/service_accounts.go:192]:
```suggestion
}

```

File: internal/kafka/test/integration/serviceaccounts_test.go [hunk 2]
```diff
@@ -270,6 +270,21 @@ func TestServiceAccounts_InputValidation(t *testing.T) {
...
+
+	h, client, teardown := test.NewKafkaHelper(t, ocmServer)
+	defer teardown()
+
+	account := h.NewRandAccount()
+	ctx := h.NewAuthenticatedContext(account, nil)
+
+	_,resp, err := client.SecurityApi.GetSsoProvider(ctx)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusOK))
```
Comment (machi1990) [inline internal/kafka/test/integration/serviceaccounts_test.go:285]: We could aslo check against the different fields in the sso_provider response body

**Review by machi1990 — APPROVED** [review]
The change lgtm.

I'd like someone involved in the sso migration effort to also have a look. 
/cc @ziccardi 

And the UI/CLI to have awareness of this new endpoint.

**Comment by akoserwal** [comment]
Do we want to make this end-point public?

**Comment by Rajagopalan-Ranganathan** [comment]
> Do we want to make this end-point unsecured? @carlesarnal @wtrocki @Rajagopalan-Ranganathan @pmuir @davidffrench

I dont think there is any sensitive information that we are leaking by making this public, as these are already public URLs.. Provided that there might be some customer or 3rd party applications that might want to use this during/for the migration effort, it should make it easier for them to have this a public end point as well. But would get the opinion of sso.rh team and @pmuir as well regarding this

**Pull request merged**